internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -1123,9 +1123,11 @@ func resourceLinuxVirtualMachineUpdate(d *pluginsdk.ResourceData, meta interface
 	if d.HasChange("license_type") {
 		shouldUpdate = true
 
-		if v, ok := d.GetOk("license_type"); ok {
-			update.Properties.LicenseType = pointer.To(v.(string))
+		licenseType := "None"
+		if v, ok := d.GetOk("license_type"); ok && v.(string) != "" {
+			licenseType = v.(string)
 		}
```

Comment:
No need to change the `if` block as the value of `license_type` won't be empty
```suggestion
		if v, ok := d.GetOk("license_type"); ok {
			update.Properties.LicenseType = pointer.To(v.(string))
		}
```